pantheon/src/main/java/tech/pegasys/pantheon/cli/subcommands/blocks/BlocksSubCommand.java

```diff
@@ -241,71 +241,127 @@ private MiningParameters getMiningParameters() {
         arity = "1..1")
     private final Long endBlock = null;
 
+    @Option(
+        names = "--format",
+        hidden = true,
+        description =
+            "The format to export, possible values are: ${COMPLETION-CANDIDATES} (default: ${DEFAULT-VALUE}).",
+        arity = "1..1")
+    private final BlockExportFormat format = BlockExportFormat.RLP;
+
     @Option(
         names = "--to",
+        required = true,
         paramLabel = MANDATORY_FILE_FORMAT_HELP,
         description = "File to write the block list instead of standard output",
         arity = "1..1")
     private File blocksExportFile = null;
 
+    @SuppressWarnings("unused")
+    @Spec
+    private CommandSpec spec;
+
     @Override
     public void run() {
 
-      LOG.info("Runs export sub command");
+      LOG.info("Export {} block data to file {}", format, blocksExportFile.toPath());
 
       checkCommand(this, startBlock, endBlock);
+      final Optional<MetricsService> metricsService = initMetrics(parentCommand);
 
-      Optional<MetricsService> metricsService = initMetrics(parentCommand);
-
+      final PantheonController<?> controller = createPantheonController();
       try {
-
-        final BlockExporter.ExportResult exportResult =
-            parentCommand.blockExporter.exportBlockchain(
-                parentCommand.parentCommand.buildController(), startBlock, endBlock);
-
-        outputBlock(exportResult.blocks);
-
-        if (exportResult.blocks.isEmpty()) {
-          throw new ExecutionException(new CommandLine(this), "No block found at the given index");
-        } else if (!exportResult.allBlocksAreFound) {
-          throw new ExecutionException(
-              new CommandLine(this),
-              "Partial export due to inability to recover all requested blocks");
+        switch (format) {
+          case RLP:
+            exportRlpFormat(controller);
+            break;
+          default:
+            throw new ParameterException(
```

Comment:
Nice, this future proofs us against random formats being added to the enum and by effect being exposed to the CLI.